File: backend/webapi/Models/SubmittingAgencyModel.cs
namespace Pidp.Models;

using NodaTime;

public class SubmittingAgencyModel
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string IdpHint { get; set; } = string.Empty;
    public Instant? ClientCertExpiry { get; set; }
    public int? LevelOfAssurance { get; set; }
    public bool HasRealm { get; set; }
    public bool HasIdentityProvider { get; set; }
    public bool HasIdentityProviderLink { get; set; }
    public List<string> Warnings { get; set; } = new List<string>();
}

File: workspace/apps/pidp/src/app/features/admin/submitting-agency/submitting-agency.model.ts
export interface SubmittingAgency {
  name: string;
  idpHint: string;
  code: string;
  levelOfAssurance: number;
  clientCertExpiry: Date;
  hasRealm: boolean;
  hasIdentityProvider: boolean;
  hasIdentityProviderLink: boolean;
  warnings: string[];
}
